modules/data/event-hub/main.bicep

```
@@ -0,0 +1,245 @@
+@description('Optional. Name for the Event Hub cluster, Alphanumerics and hyphens characters, Start with letter, End with letter or number.')
+param clusterName string = 'null'
+
+@description('Optional. The quantity of Event Hubs Cluster Capacity Units contained in this cluster.')
+param clusterCapacity int = 1
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('''Optional. The name of the event hub namespace to be created.
+Below paramters you can pass while creating the Azure Event Hub namespace.
+sku: (Optional) Possible values are "Basic" or "Standard" or "Premium". Detault to 'Standard'.
+capacity: (Optional) int, The Event Hubs throughput units for Basic or Standard tiers, where value should be 0 to 20 throughput units.The Event Hubs premium units for Premium tier, where value should be 0 to 10 premium units. Default to 1.
+zoneRedundant: (Optional) bool, Enabling this property creates a Standard Event Hubs Namespace in regions supported availability zones. Default to false.
+isAutoInflateEnabled: (Optional) bool,  Value that indicates whether AutoInflate is enabled for eventhub namespace. Only available for "Standard" sku. Default to false.
+maximumThroughputUnits: (Optional) int, Upper limit of throughput units when AutoInflate is enabled, value should be within 0 to 20 throughput units. ( '0' if AutoInflateEnabled = true)
+disableLocalAuth: (Optional) bool, This property disables SAS authentication for the Event Hubs namespace. Default to false.
+kafkaEnabled: (Optional) bool, Value that indicates whether Kafka is enabled for eventhub namespace. Default to true.
+''')
+param eventHubNamespaces object = {}
```

I think we have to many optional parameters here.

If I deploy the deployment with just the location parameter, I should end up with the minimum set of resources needed. But I think with how things are right now, it would end up deploying nothing. Let's give these a default value so they deploy at least one set of resources, and it will also help users understand the expected schema.

```bicep
module eventhubnamespace 'br/public:data/event-hub:1.0.0' = {
  name: '${uniqueString(deployment().name, 'eastus')}-event-hub-namespace'
  params: {
    location: 'eastus'
  }
}
```

```suggestion
param eventHubNamespaces object = {
  evhns: {
      sku: 'Standard'
      capacity:  4
      maximumThroughputUnits: 2
      zoneRedundant: true
      isAutoInflateEnabled: true
  }
}
```

---

Great idea, Done